code/ui/components/src/new/Button/Button.tsx

```diff
@@ -4,18 +4,30 @@ import { styled } from '@storybook/theming';
 import { darken, lighten, rgba, transparentize } from 'polished';
 import type { PropsOf } from '../utils/types';
 
-export interface ButtonProps<T extends React.ElementType = React.ElementType> {
+interface CommonProps<T extends React.ElementType = React.ElementType> {
   as?: T;
-  children: string;
   size?: 'small' | 'medium';
   variant?: 'primary' | 'secondary' | 'tertiary';
-  icon?: ReactNode;
-  iconOnly?: boolean;
   onClick?: () => void;
   disabled?: boolean;
   active?: boolean;
 }
 
+type ButtonIconOnlyProps = {
+  iconOnly?: true;
```

Comment:
I think this should not be optional in this case

Reply:
Thanks. I'll make it mandatory.